tests/acceptance/standard/test_files.py

```diff
@@ -70,32 +73,30 @@ def test__set_file_specific_branch(self, gitlab, group_and_project, branch):
                 content: "Content for {branch} only"
         """
 
-        run_gitlabform(set_file_specific_branch, group_and_project)
+        run_gitlabform(set_file_specific_branch, project.path_with_namespace)
 
-        commit = gitlab.get_last_commit(group_and_project, branch)
+        the_branch = project.branches.get(branch)
         assert all(
-            [text in commit["message"] for text in ["Automated", "made by gitlabform"]]
+            [
+                text in the_branch.commit["message"]
+                for text in ["Automated", "made by gitlabform"]
+            ]
         )
 
-        file_content = gitlab.get_file(group_and_project, branch, "README.md")
-        assert file_content == f"Content for {branch} only"
+        project_file = project.files.get(ref=branch, file_path="README.md")
+        assert project_file.decode().decode("utf-8") == f"Content for {branch} only"
```

Comment:
I realize this is a slightly awkward interface (decoding from base64, then to string), which you do in your wrapper helper. I've added a small PR to make this a bit nicer in the future:
https://github.com/python-gitlab/python-gitlab/pull/2396

Reply:
Thanks for that PR. :) I learned from it about `@overload` and `...`. 😅